docs/plugin_developers_guide/02_writing_plugins.rst

```diff
@@ -425,3 +433,22 @@ The code that connects to the device should then look at the *discovered* config
 The example here was written in C++, there is nothing that is specific to C++ however and the same approach can be taken in Python.
 
 One thing to note however, the *plugin_info* call is used in the display of available plugins, discovery code that is very slow will impact the performance of plugin selection.
+
+Writing Audit Trail
+~~~~~~~~~~~~~~~~~~~
+
+Plugins are able to write records to the audit trail. These records must use one of the predefined audit code that are support by the system. See |audit_trail| for details of the supported audit codes within the system.
+
+In C++ you use the *AuditLogger* class to write these audit trail entries, this is a singleton object that is access vi the getLogger method.
+
+.. code-block:: C
+
+   AuditLogger *audit = AuditLogger::getLogger();
+   audit->audit("NHDWN", "INFORMATION");
+
+There is also a convenience function that can be used if you not not want to define a local pointer the AuditLogger
+
+.. code-block:: C
+
+   AuditLogger::auditLog("NHAVL", "INFORMATION");
+
```

Comment:
What about audit logger in Python?

Reply:
We don't have that yet. There is a Jira open for it FOGL-7429.